python/graphstorm/model/utils.py

```diff
@@ -703,8 +703,111 @@ def shuffle_predict(predictions, id_mapping_file, pred_type,
                 len(predictions)).cpu() # predictions are stored in CPU
     return predictions[local_id_mapping]
 
+class NodeIDShuffler():
+    """ Shuffle node ids into the original node ids according to node_id_mappings
+
+        Parameters
+        ----------
+        g: Dist DGLGraph
+            Graph.
+        node_id_mapping_file:
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        ntypes: list of str
+            The node types that will have node ids shuffled.
+    """
+    def __init__(self, g, node_id_mapping_file, ntypes=None):
```

Comment:
i think users need to provide `ntypes`. otherwise, the code probably doesn't work.